azavea/raster-foundry
# Add support for non-Base64 encoded JWT client secrets

## Overview

Auth0 recently made a change to the way they encode their client secrets. Previously, they were Base64 encoded, but now they are not.

In an effort to better support non-Base64 encoded client secrets in Scala, I looked for a higher quality JWT library, which led me to `jwt-core` and `jwt-json4s-jackson`.

For Python, I just removed the Base64 decoding operation.

These changes aim to both replace the previous `choffmeister` JWT library and also allow the application to support non-Base64 encoded Auth0 secrets.

Fixes https://github.com/azavea/raster-foundry/issues/1039

## Testing Instructions

First, comment out the following lines in `bootstrap` to ensure that you don't override the local `.env` file with the one from S3:

```diff
diff --git a/scripts/bootstrap b/scripts/bootstrap
index 3483fef9..459f400c 100755
--- a/scripts/bootstrap
+++ b/scripts/bootstrap
@@ -16,11 +16,11 @@ Example: ./scripts/bootstrap
 
 function main() {
 
-    echo "Pulling down development environment..."
-    pushd "${DIR}/.."
-    # Download environment configuration from S3
-    aws s3 cp "s3://rasterfoundry-development-config-us-east-1/.env" ".env"
-    popd
+    # echo "Pulling down development environment..."
+    # pushd "${DIR}/.."
+    # # Download environment configuration from S3
+    # aws s3 cp "s3://rasterfoundry-development-config-us-east-1/.env" ".env"
+    # popd
 
     echo "Building/Pulling containers..."
     docker-compose \
```

From there, you'll need to update the `.env` file so that it contains the appropriate Auth0 configuration parameters for `raster-foundry-dev.auth0.com` (I can give them to you out-of-band).

Lastly, I tested these changes by executing the test suite locally and [interacting with the application server and tile server](http://localhost:9100/):

```bash
vagrant@vagrant-ubuntu-trusty-64:/opt/raster-foundry$ ./scripts/test
vagrant@vagrant-ubuntu-trusty-64:/opt/raster-foundry$ ./scripts/server
```
